lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -4,109 +4,136 @@ module Hyacinth
-            log(:error, "#mint_and_store_doi: EZID API call to mint_identifier was unsuccessful.")
-            log(:error, "#mint_and_store_doi: Response from EZID server follows:\n#{http_error_message(response)}")
-            return nil
+          datacite_data = Datacite::RestApi::V2::Data.new
+          datacite_data.prefix = DATACITE[:prefix]
+          map_metadata(hyacinth_metadata, datacite_data, target_url) if hyacinth_metadata
+          datacite_data.build_mint(doi_state, hyacinth_metadata.present?)
+          rest_api = Datacite::RestApi::V2::Api.new(DATACITE[:rest_api], DATACITE[:user], DATACITE[:password])
+          rest_api_response = rest_api.post_dois datacite_data.generate_json_payload
+          unless rest_api_response.status.eql? 201
+            Rails.logger.error "Did not mint a DOI! Response Code: #{rest_api_response.status}." \
+                               "Response Body: #{rest_api_response.body}." \
+                               "Request Body: #{rest_api.most_recent_request_body}"
+            return
           end
-          ezid_doi_instance.identifier.sub(/^doi\:/, '')
-        end
-
-        # Returns true if an identifier exists in the external management system
-        def exists?(_id)
-          raise NotImplementedError
+          rest_api.parse_doi_from_api_response_body(rest_api_response.body)
         end
 
-        # @param id [String]
-        # @param digital_object [DigitalObject]
-        # @param location_uri [String]
-        # @return [Boolean] true if this adapter can handle this type of identifier
-        def update_impl(id, digital_object, location_uri)
-          return update_doi_target_url(id, location_uri) if digital_object.blank?
-
-          # get the metadata from hyacinth
-          hyacinth_metadata = Datacite::Metadata.new digital_object.as_json
-          # prepare the metadata into an acceptable format for EZID
-          datacite_metadata = Datacite::MetadataBuilder.new hyacinth_metadata
-          # ApiSession#modify_identifier returns true if the response from the EZID server indicated
-          # success, else it returns false
-          opts = { datacite: datacite_metadata.datacite_xml, _status: IDENTIFIER_STATUS[:public] }
-          opts[:_target] = location_uri unless location_uri.blank?
-          api_session.modify_identifier(id, opts)
+        # return nil if unsuccessful, else returns the response body
+        def update_doi(doi,
+                       hyacinth_metadata, # Hyacinth::Adapters::ExternalIdentifierAdapter::Datacite::Metadata
+                       target_url, doi_state = nil)
+          datacite_data = Datacite::RestApi::V2::Data.new
+          datacite_data.prefix = DATACITE[:prefix]
+          map_metadata(hyacinth_metadata, datacite_data, target_url)
+          datacite_data.build_properties_update(doi_state)
+          rest_api = Datacite::RestApi::V2::Api.new(DATACITE[:rest_api], DATACITE[:user], DATACITE[:password])
```

Comment:
Same comment here about `DATACITE` global variable usage.

Reply:
DATACITE global variable is not referenced anymore in the Datacite class. Instead, the adapter_config parameter passed into the initialize method is used to populate pertinent instance variables.